boring/src/ssl/mod.rs

```diff
@@ -1560,6 +1629,33 @@ pub struct CipherBits {
     pub algorithm: i32,
 }
 
+#[repr(transparent)]
+pub struct ClientHello(ffi::SSL_CLIENT_HELLO);
+
+impl ClientHello {
+    /// Returns the data of a given extension, if present.
+    ///
+    /// This corresponds to [`SSL_early_callback_ctx_extension_get`].
+    ///
+    /// [`SSL_early_callback_ctx_extension_get`]: https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_early_callback_ctx_extension_get
+    pub fn get_extension(&self, type_: ExtensionType) -> Option<&[u8]> {
+        let slice = unsafe {
+            let mut ptr = ptr::null();
+            let mut len = 0;
+            let result =
+                ffi::SSL_early_callback_ctx_extension_get(&self.0, type_.0, &mut ptr, &mut len);
```

Comment:
Ah, it's `const uint8_t **`. I thought it's `uint8_t *` which we need to de-alloc ourselves.